web/js/mapUI/components/tile-errors/tile-error-handler.js

```diff
@@ -0,0 +1,171 @@
+
+  if (hasSubdailyLayers) {
+    const timestamp = `${year}-${month}-${day}T${hours}:${minutes}:${seconds}`;
+    const adjustedTimezone = convertTimestamp(timestamp);
+    return adjustedTimezone;
+  }
+
+  return `${year}-${month}-${day}T00:00:00`;
+}
+
+// gets a week ago from real time date
+function weekAgo(realTimeDate) {
+  const inputDate = new Date(realTimeDate);
+  const oneWeekInMilliseconds = 7 * 24 * 60 * 60 * 1000;
+  const earlierDate = new Date(inputDate.getTime() - oneWeekInMilliseconds);
+  const earlierDateString = earlierDate.toString();
+  return earlierDateString;
+}
+
+// safe gaurd agaisnt stepping back too far, we only want to go back 7 days for daily and 24 hours for subdaily
+function compareDailyDates(lastDateToCheck, selectedDate) {
+  const lastDate = new Date(lastDateToCheck);
+  const selected = new Date(selectedDate);
+  lastDate.setHours(0, 0, 0, 0);
+  selected.setHours(0, 0, 0, 0);
+  return selected > lastDate;
+}
+
+// safe gaurd agaisnt stepping back too far, we only want to go back 23 hours for subdaily
+function compareSubdailyDates(lastDateToCheck, selectedDate) {
+  const lastDate = new Date(lastDateToCheck);
+  const selected = new Date(selectedDate);
+  const lastDateHour = lastDate.getHours();
+  const selectedDateHour = selected.getHours();
+  // check if the hour value in selectedDate is exactly one hour more than lastDateToCheck
+  // also, handle the case where selectedDateHour is 0 and lastDateHour is 23
+  if (selectedDateHour === (lastDateHour + 1) % 24) {
+    return false;
+  }
+  return true;
+}
+
+function TileErrorHandler({ action }) {
+  const dispatch = useDispatch();
+  const clearErrorTiles = () => { dispatch(clearErrorTilesAction()); };
+  const selectDate = (date) => { dispatch(selectDateAction(date)); };
+  const selectInterval = (delta, timeScale, customSelected) => { dispatch(selectIntervalAction(delta, timeScale, customSelected)); };
+
+  const {
+    isKioskModeActive, errorTiles, selectedDate, date, compare, isLoading, realTimeDate,
+  } = useSelector((state) => ({
+    isKioskModeActive: state.ui.isKioskModeActive,
+    errorTiles: state.ui.errorTiles,
+    selectedDate: state.date.selected,
+    date: state.date,
+    compare: state.compare,
+    isLoading: state.loading.isLoading,
+    realTimeDate: state.date.appNow,
+  }));
+  const hasSubdailyLayers = useSelector((state) => subdailyLayersActive(state));
+
+  // 7 days ago from real time date
+  const lastDateToCheck = weekAgo(realTimeDate);
+
+  // true === safe (date is younger than last date to check)
+  const dailySafeguardCheck = compareDailyDates(lastDateToCheck, selectedDate);
+  const hourlySafeguardCheck = compareSubdailyDates(lastDateToCheck, selectedDate);
+
+  useEffect(() => {
+    if (isKioskModeActive && errorTiles.length && dailySafeguardCheck && hourlySafeguardCheck && !isLoading) {
+      sortErrorTiles();
+    }
+  }, [action]);
+
+  // sorting tiles for daily & subdaily
+  function sortErrorTiles() {
```

Comment:
Are there perf concerns here? Would we be better off storing separate daily/subdaily in state & avoid sorting all error tiles each time?